addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/parser/CommonRpcParser.java

```diff
@@ -121,4 +125,66 @@ protected Object convertToType(HmDatapoint dp, Object value) {
         }
     }
 
+    /**
+     * Assembles a datapoint with the given parameters.
+     */
+    protected HmDatapoint assembleDatapoint(String name, String unit, String type, String[] options, Object min,
+            Object max, Integer operations, Object defaultValue, HmParamsetType paramsetType, boolean isHmIpDevice)
+            throws IOException {
+        HmDatapoint dp = new HmDatapoint();
+        dp.setName(name);
+        dp.setDescription(name);
+        dp.setUnit(StringUtils.replace(StringUtils.trimToNull(unit), "�", "°"));
```

Comment:
Wouldn't a proper charset conversion be the cleaner solution here? Seeing an "unknown character" here in the diff makes me wonder, what might happen as soon as the next person touches the code...